plugins/woocommerce/src/Blocks/Utils/ProductCollectionUtils.php

```diff
@@ -78,6 +79,87 @@ public static function remove_query_array( $queries, $key, $value ) {
 		return self::remove_empty_array_recursive( $queries );
 	}
 
+	/**
+	 * Parse WP Query's front-end context for the Product Collection block.
+	 *
+	 * The sourceData structure depends on the context type as follows:
+	 * - site:    [ ]
+	 * - order:   [ 'orderId'    => int ]
+	 * - cart:    [ 'productIds' => int[] ]
+	 * - archive: [ 'taxonomy'   => string, 'termId' => int ]
+	 * - product: [ 'productId'  => int ]
+	 *
+	 * @return array $context {
+	 *     @type string  $type        The context type. Possible values are 'site', 'order', 'cart', 'archive', 'product'.
+	 *     @type array   $sourceData  The context source data. Can be the product ID of the viewed product, the order ID of the current order, etc.
+	 * }
+	 */
+	public static function parse_frontend_location_context() {
+		global $wp_query;
+
+		// Default context.
+		// Hint: The Shop page uses the default context.
+		$type        = 'site';
+		$source_data = array();
+
+		if ( ! ( $wp_query instanceof WP_Query ) ) {
+
```

Comment:
```suggestion
```

Reply:
I'll opt to leave the spaces for readability. Thoughts?